reana_job_controller/kubernetes_job_manager.py

```diff
@@ -311,9 +313,6 @@ def _add_krb5_init_container(self, secrets_volume_mount):
             "name": current_app.config["KRB5_CONTAINER_NAME"],
             "imagePullPolicy": "IfNotPresent",
             "volumeMounts": [secrets_volume_mount] + volume_mounts,
-            "security_context": client.V1PodSecurityContext(
-                run_as_group=WORKFLOW_RUNTIME_USER_GID, run_as_user=self.kubernetes_uid
-            ),
         }
```

Comment:
Testing Kerberos use case, I get the following permissions for a workflow asking for uid=1000 and kerberos=true:

```
$ ls -l /krb5_cache/krb5_1000
-rw------- 1 1000 root 4338 Jul  3 14:22 /krb5_cache/krb5_1000
$ $ ls -lR /etc/reana/secrets/
/etc/reana/secrets/:
total 0
lrwxrwxrwx 1 root root 19 Jul  3 14:21 johndoe.keytab -> ..data/johndoe.keytab
$ ls -lR /etc/reana/secrets/..data/johndoe.keytab
-rw-r--r-- 1 root root 128 Jul  3 14:21 /etc/reana/secrets/..data/johndoe.keytab
```